File: week08/tags.pl
#!/usr/bin/perl

$url = ARGV[0];

$source = `wget -q -O- '$url'` or die;
$source = lc $website;
$source =~ s/<!--.*?-->//g;

@tags = $source =~ /<[\s\/]*(\w+)/g;
foreach $tag (@tags) {
    $tagcount{$tag}++;
}

if ($sort_by_freq) {
    @tags = sort {$tagcount{$a} <=> $tagcount{$b}} keys %tagcount;
} else {
    @tags = sort keys %tagcount;
}

print "$_ $tagcount{$_}\n" foreach (@tags);

File: week08/tags.py
#!/usr/bin/env python

import sys, re, subprocess

url = sys.argv[1];

wget = subprocess.Popen(['wget','-q','-O-',url], stdout=subprocess.PIPE)
source = wget.communicate()[0].rstrip('\n')
source = re.sub("<!--.*?-->", "", source)
tags = dict()

for tag in re.findall("<[\s\/]*(\w+)", source):
    tag = tag.lower()
    tags.setdefault(tag, 0)
    tags[tag] += 1;

for tag in sorted(tags.keys()):
    print tag, tags[tag]
